### PR TITLE
Fixed index mapping of terms and vocabularies

### LINKED ISSUE #779: Vocabularies with spaces and without terms show up incorrectly in listing
The vocabulary slug is shown instead of the vocabulary name. It looks like this is due to a for loop over terms in `convert_aggregate()` which will skip vocabularies that don't contain any terms.

### LINKED ISSUE #785: Renamed term doesn't update mapping properly
To reproduce, add a term, assign a term to a resource, then rename the term. There are a couple things to notice:
- the old term name is used in the facet counts
- When searching the index only knows about the old term, resources are only updated if manually resaved.

One way to fix this is to use term ids as values for the mapping instead of slugs since they don't change

### LINKED ISSUE #786: Editing a vocabulary doesn't update mapping properly
Unlike in #785 you do see the updated vocabulary name in the facets but we only reindex resources where we would have removed terms from the vocabulary, so you will see that the new name doesn't have any facet counts for its terms even though they should exist.

If we used vocabulary ids here, something like 'vocab_1' for the index key, I think this problem would be solved since ids are immutable.

### PR DESCRIPTION
**Pull request by noisecapella**
Fixes #786 
Fixes #785 
Fixes #779

### CODE REVIEW

File: rest/tests/test_vocabulary.py [hunk 1]
```diff
@@ -896,6 +896,134 @@ def test_vocab_num_queries(self):
...
+        vocab_id = vocab_result['id']
+        vocab_key = make_vocab_key(vocab_id)
+
+        self.assertEqual(
+            self.get_results()['facet_counts'][vocab_key]['facet']['label'],
+            'name with spaces'
+        )
+
+        self.create_term(self.repo.slug, vocab_slug)
+        self.assertEqual(
```
Comment (ShawnMilo) [inline rest/tests/test_vocabulary.py]: What is this assertion supposed to test? It's the same as the one just above (line 1016), and I'm missing how adding a term could change the output of `get_results()`.

**Comment by ShawnMilo** [comment]
Very minor nit: I think `"brand new name"` should be a variable instead of repeating the string.

**Comment by ShawnMilo** [comment]
One tiny nit and one question. The rest looks good.

File: rest/tests/test_vocabulary.py (hunk 1 above)
Comment (noisecapella) [inline rest/tests/test_vocabulary.py]: There was a bug where if a vocab had no terms the vocab slug would show up in place of the vocab label (#779)

Comment (ShawnMilo) [inline rest/tests/test_vocabulary.py]: Okay, I didn't make the association. Thanks. It's good to merge, if you want to ignore the nit.

Comment (noisecapella) [inline rest/tests/test_vocabulary.py]: Oh, I guess this assertion doesn't test that, it's the one above that does.

**Pull request merged**